napari_btrack/track.py

```diff
@@ -31,23 +151,276 @@ def run_tracker(objects, config_file_path):
+    """Helper function to update a container's widgets
+    with the values in a given tracker config.
+    """
+    for model in ["motion_model", "hypothesis_model", "object_model"]:
+        model_config = getattr(config, model)
+        if model_config:
+            for parameter, value in model_config:
+                if parameter in hidden_variable_names:
+                    continue
+                if parameter in Matrices().names:
+                    sigma = Matrices.get_sigma(parameter, value)
+                    getattr(container, f"{parameter}_sigma").value = sigma
+                if parameter == "hypotheses":
+                    for hypothesis in all_hypotheses:
+                        getattr(container, hypothesis).value = False
+                    for hypothesis in value:
+                        getattr(container, hypothesis).value = True
+                else:
+                    getattr(container, parameter).value = value
+    mode_is_cell = config.motion_model.A[0, 3] == 1
+    print("mode is cell: ", mode_is_cell)
+    container.mode.value = "cell" if mode_is_cell else "particle"
+
+
+def _create_button_widgets(widgets: List[Widget]) -> None:
+    """Create the set of button widgets needed:
+    run, save/load configuration and reset."""
+    widget_names = [
+        "load_config_button",
+        "save_config_button",
+        "reset_button",
+        "call_button",
+    ]
+    widget_labels = [
+        "Load configuration",
+        "Save configuration",
+        "Reset defaults",
+        "Run",
+    ]
+    widgets.extend([create_widget(**html_label_widget("Control buttons"))])
+    widgets.extend(
+        [
+            create_widget(name=widget_name, label=widget_label, widget_type=PushButton)
+            for widget_name, widget_label in zip(widget_names, widget_labels)
+        ]
+    )
+
+
+def track() -> Container:
+    """
+    Create a series of widgets programatically
+    """
+    # initialise a list for all widgets
+    widgets: list = []
+
+    # the different model types
+    default_model_configs = [
+        default_cell_config.motion_model,
+        default_cell_config.object_model,
+        default_cell_config.hypothesis_model,
+    ]
+
+    # create all the widgets
+    _create_napari_specific_widgets(widgets)
+    _create_cell_or_particle_widget(widgets)
+    _create_pydantic_default_widgets(widgets, default_model_configs)
+    _create_button_widgets(widgets)
+
+    btrack_widget = Container(widgets=widgets)
+
+    @btrack_widget.reset_button.changed.connect
+    def restore_defaults() -> None:
+        _tracker_config_to_widgets(btrack_widget, default_cell_config)
+
+    @btrack_widget.call_button.changed.connect
+    def run() -> None:
+        config = _widgets_to_tracker_config(btrack_widget)
+        segmentation = btrack_widget.segmentation.value
+        data, properties, graph = run_tracker(segmentation, config)
```

Comment:
Should this be run in a new thread? (I'm guessing yes, but probably beyond scope of this PR)

Reply:
Indeed. See #17.